_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1041,11 +1041,85 @@ private function _process_onboarding( $data ) {
 			}
 		}
 
+		// Add/update business address widget
+		//
+		if ( isset( $data['businessInfo'] ) {
+			$first_sidebar = self::get_first_sidebar();
+			$widget_options = ''//our input data 
+
+			if ( $first_sidebar ) {
+					if ( ! self::have_contact_info_widget( $first_sidebar ) ) {
+					self::insert_widget_in_sidebar( 'widget_contact_info', $widget_options, $first_sidebar );
+				} else {
+					self::update_widget_in_sidebar( 'widget_contact_info', $widget_options, $first_sidebar );
+				}
+			}
+		}
+
 		return empty( $error )
 			? ''
 			: join( ', ', $error );
 	}
 
+	/**
+	 * Inserting widget into the sidebar
+	 */
+	 static function insert_widget_in_sidebar( $widget_id, $widget_options, $sidebar ) {
+		// Retrieve sidebars, widgets and their instances
+		$sidebars_widgets = get_option( 'sidebars_widgets', array() );
+		$widget_instances = get_option( 'widget_' . $widget_id, array() );
+		// Retrieve the key of the next widget instance
+		$numeric_keys = array_filter( array_keys( $widget_instances ), 'is_int' );
+		$next_key = $numeric_keys ? max( $numeric_keys ) + 1 : 2;
+		// Add this widget to the sidebar
+		if ( ! isset( $sidebars_widgets[ $sidebar ] ) ) {
+			$sidebars_widgets[ $sidebar ] = array();
+		}
+		$sidebars_widgets[ $sidebar ][] = $widget_id . '-' . $next_key;
+		// Add the new widget instance
+		$widget_instances[ $next_key ] = $widget_options;
+		// Store updated sidebars, widgets and their instances
+		update_option( 'sidebars_widgets', $sidebars_widgets );
+		update_option( 'widget_' . $widget_id, $widget_instances );
+	}
+
+		static function update_widget_in_sidebar( $widget_id, $widget_options, $sidebar ) {
+		// Retrieve sidebars, widgets and their instances
+		$sidebars_widgets = get_option( 'sidebars_widgets', array() );
+		$widget_instances = get_option( 'widget_' . $widget_id, array() );
+		// Retrieve index of first widget instance in that sidebar
+		$widget_key = false;
+		foreach ( $sidebars_widgets[ $sidebar ] as $widget ) {
+			if ( strpos( $widget, 'widget_contact_info' ) !== false ) {
+				$widget_key = absint( str_replace( 'widget_contact_info-', '', $widget ) );
+				break;
+			}
+		}
+		if ( ! $widget_key ) {
+			return;
+		}
+		// Update the widget instance with the new data
+		$widget_instances[ $widget_key ] = array_merge( $widget_instances[ $widget_key ], $widget_options );
+		// Store updated widget instances
+		update_option( 'widget_' . $widget_id, $widget_instances );
+	}
+
+	static function get_first_sidebar() {
+		$active_sidebars = get_option( 'sidebars_widgets', array() );
+		$excluded_keys = array(
+			'wp_inactive_widgets',
+			'array_version',
+		);
+		foreach ( $excluded_keys as $key ) {
+			if ( isset( $active_sidebars[ $key ] ) ) {
+				unset( $active_sidebars[ $key ] );
+			}
+		}
```

Comment:
Since there are only two keys, the lines from sidebar option fetch up to the loop could be written as
```php
$active_sidebars = get_option( 'sidebars_widgets', array() );
unset( $active_sidebars[ 'wp_inactive_widgets' ], $active_sidebars[ 'array_version' ] );
```
The `unset` won't fail if the array item is not set.

It could also be written with an uglier one liner that ends up being longer :D
```php
$active_sidebars = array_diff_key(
    get_option( 'sidebars_widgets', array() ),
    array_flip( array( 'wp_inactive_widgets', 'array_version' ) )
);
```
or

Reply:
Thanks for the review :) I settled for the first option to aid readability 👍

Reply:
Using `Jetpack_Widgets` does not allow 'wp_inactive_widgets', so there is in fact a single key :)